python/HH_SF.py

```diff
@@ -61,4 +61,78 @@ def HHbtag_SFRDF(**kwargs):
 
     """
     year = kwargs.pop("year")
-    return lambda: HHbtag_SFRDFProducer(year, **kwargs)
\ No newline at end of file
+    return lambda: HHbtag_SFRDFProducer(year, **kwargs)
+
+class HH_pNetSFRDFProduced(JetLepMetSyst):
+    def __init__(self, *args, **kwargs):
+        super(HH_pNetSFRDFProduced, self).__init__(*args, **kwargs)
+        self.isMC = kwargs.pop("isMC")
+        self.year = kwargs.pop("year")
+
+        # Check whether HH-like, TT-like, DY-like or other for pNet SF:
+        #  - HH-like : ggHH, VBFHH, ZH, WH, ttH, ggH, qqH, ttWH, ttZH
+        #  - TT-like : TT fullyLep, semiLep, fullyHad
+        #  - DY_like : all DY samples
+        #  - other   : all other samples
+        dsetName = kwargs.pop("dataset")
+        if "HH" in dsetName or "Hto2B" in dsetName or "WH" in dsetName or "ZH" in dsetName:
+            sampleType = "HHlike"
+        elif "TTto" in dsetName:
+            sampleType = "TTlike"
+        elif "DY" in dsetName:
+            sampleType = "DYlike"
+        else:
+            sampleType = "other"
+        self.sampleType = sampleType
```

Comment:
As already noted by @tcuisset during the CCLUB meeting (and duly noted in https://gitlab.cern.ch/cclubbtautau/AnalysisCore/-/issues/2), this is not really the ideal way of splitting the samples, for two reasons:
 - sample names might change year from year (and even between different productions)
 - this splitting works okay-ish for HH, but might not be the ideal one for the ZH/ZZ